plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RollbackToSnapshotProcedure.java

```diff
@@ -64,8 +67,10 @@ public Procedure get()
 
     public void rollbackToSnapshot(ConnectorSession clientSession, String schema, String table, Long snapshotId)
     {
-        SchemaTableName schemaTableName = new SchemaTableName(schema, table);
-        Table icebergTable = catalogFactory.create(clientSession.getIdentity()).loadTable(clientSession, schemaTableName);
-        icebergTable.rollback().toSnapshotId(snapshotId).commit();
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
```

Comment:
Just wondering, pls do not do any implementation yet: considering that this is a fix for the `rollback_to_snapshot` stored procedure call, shouldn't we then apply this method to other places as well in `trino-iceberg` ?

Reply:
Probably there may be other places where this is necessary but it is not that easy to spot them. 
Our tests cover a lot of scenarios and they seem to work. 
Also not only trino-iceberg is affected. Actually every place that calls `newEmptyConfiguration` can be a problem